Mathlib/MeasureTheory/Function/UnifTight.lean

```diff
@@ -0,0 +1,514 @@
+    intro M
+    apply measurableSet_lt hmf measurable_const
+  have hmeas : ∀ M : ℕ, Measurable ({ x | f x < 1 / (↑M + 1) }.indicator f) := by
+    intro M
+    apply hmf.indicator
+    apply hmeas_lt M
+  -- show that the sequence a.e. converges to 0
+  have htendsto :
+      ∀ᵐ x ∂μ, Tendsto (fun M : ℕ => { x | f x < 1 / (↑M + 1) }.indicator f x) atTop (𝓝 0) :=
+    univ_mem' (id fun x => tendsto_ENNReal_indicator_lt f x)
+  -- use Lebesgue dominated convergence to show that the integrals eventually go to zero
+  have : Tendsto (fun n : ℕ ↦ ∫⁻ a, { x | f x < 1 / (↑n + 1) }.indicator f a ∂μ)
+      atTop (𝓝 (∫⁻ (_ : α), 0 ∂μ)) := by
+    refine' tendsto_lintegral_of_dominated_convergence _ hmeas _ hf.ne htendsto
+    -- show that the sequence is bounded by f (which is integrable)
+    refine' fun n => univ_mem' (id fun x => _)
+    by_cases hx : f x < 1 / (↑n + 1)
+    · dsimp
+      rwa [Set.indicator_of_mem]
+    · dsimp
+      rw [Set.indicator_of_not_mem]
+      · exact zero_le _
+      · assumption
+  -- rewrite limit to be more usable and get the sufficiently large M, so the integral is < ε
+  rw [lintegral_zero, ENNReal.tendsto_atTop_zero] at this
+  obtain ⟨M, hM⟩ := this (ENNReal.ofReal ε) (ENNReal.ofReal_pos.2 hε)
+  simp only [true_and_iff, ge_iff_le, zero_tsub, zero_le, sub_zero, zero_add, coe_nnnorm,
+    Set.mem_Icc] at hM
+  -- the target estimate is now in hM
+  have hM := hM M le_rfl
+  -- let s be the complement of the integration domain in hM,
+  -- prove its measurability and finite measure
+  have : { x | f x < 1 / (↑M + 1) } = { x | 1 / (↑M + 1) ≤ f x }ᶜ := by
+    apply Set.ext; intro x
+    simp only [mem_compl_iff, mem_setOf_eq, not_le]
+  have hms := (hmeas_lt M).compl
+  rw [this] at hM hms
+  rw [compl_compl] at hms
+  have hμs := calc
+    μ { x | 1 / (↑M + 1) ≤ f x }
+      ≤ (∫⁻ a, f a ∂μ) / (1 / (↑M + 1)) :=
+        meas_ge_le_lintegral_div hmf.aemeasurable (by norm_num) (by norm_num)
+    _ < ∞ := by apply div_lt_top hf.ne (by norm_num)
+  set s := { x | 1 / (↑M + 1) ≤ f x }
+  -- fulfill the goal
+  use s, hms, hμs, hM
+
+/-- A single function that is `Memℒp f p μ` is tight wrt to `μ`. -/
+theorem Memℒp.snorm_indicator_compl_le (hp_one : 1 ≤ p) (hp_top : p ≠ ∞)
+    {f : α → β} (hf : Memℒp f p μ)
+    {ε : ℝ} (hε : 0 < ε) :
+    ∃ (s : Set α) (_ : MeasurableSet s) (_ : μ s < ∞),
+      snorm (sᶜ.indicator f) p μ ≤ ENNReal.ofReal ε := by
+  -- The proof unwraps `Memℒp f p μ` and applies the analogous result for `lintegral`.
+  -- do some arithmetic that will come in useful
+  have hp_pos := zero_lt_one.trans_le hp_one
+  have hp_nz := hp_pos.ne'
+  have hrp_pos : 0 < p.toReal := ENNReal.toReal_pos hp_nz hp_top
+  have hεp : 0 < ε ^ p.toReal := by simp only [Real.rpow_pos_of_pos, hε]
+  -- decode Memℒp into a.e. strong measurability and finite snorm
+  obtain ⟨haesmf, hsnf⟩ := hf
+  -- transform snorm to lintegral
+  rw [snorm_eq_snorm' (by assumption) (by assumption)] at hsnf
```

Comment:
You probably should use [MeasureTheory.snorm_eq_lintegral_rpow_nnnorm](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Function/LpSeminorm/Basic.html#MeasureTheory.snorm_eq_lintegral_rpow_nnnorm). In general, `snorm'` is not meant to be used outside of the file where it's defined as an intermediate step towards snorm.

Reply:
Done in next update.